storage/innobase/dict/dict0dd.cc

```diff
@@ -1007,8 +1007,8 @@ dict_table_t *dd_table_create_on_dd_obj(const dd::Table *dd_table,
     uint64 trx_id = 0;
     dd::Object_id index_space_id = dd_index->tablespace_id();
 
-    if (dd_table->tablespace_id() == dict_sys_t::s_dd_space_id) {
-      sid = dict_sys_t::s_space_id;
+    if (dd_table->tablespace_id() == dict_sys_t::s_dd_dict_space_id) {
+      sid = dict_sys_t::s_dd_dict_space_id;
```

Comment:
sid should be dict_sys_t::s_dict_space_id (no DD here )

Reply:
@satya-bodapati 
```
dict_sys_t::s_dd_space_id -> dict_sys_t::s_dd_dict_space_id
dict_sys_t::s_space_id -> dict_sys_t::s_dict_space_id
```

Seems like the line in the if branch is correct, however, the sdi attribution needs to be to `sid = dict_sys_t::s_dict_space_id;`

The one that was causing the problem and @rahulmalik87 figured out was inside `dict_load_from_spaces_sdi` at `xtrabackup.cc`